module/VuFind/src/VuFind/ILS/Driver/SierraRest.php

```diff
@@ -914,32 +935,68 @@ public function updateHolds(
     ): array {
         $results = [];
         foreach ($holdsDetails as $requestId) {
+            // Fetch existing hold status:
+            $reqFields = 'status,pickupLocation,frozen'
+                . (isset($fields['frozen']) ? ',canFreeze' : '');
+            $hold = $this->makeRequest(
+                [$this->apiBase, 'patrons', 'holds', $requestId],
+                [
+                    'fields' => $reqFields
+                ],
+                'GET',
+                $patron
+            );
+            $available = in_array($hold['status']['code'], ['b', 'j', 'i']);
```

Comment:
Should the available codes be defined in a property for easier overriding/future configuration support?

Reply:
Those are pretty static, but since they're used in a couple of places, that makes sense. Now done.